core/matrix/bccoo.cpp

```diff
@@ -0,0 +1,774 @@
+        auto host_bccoo = Bccoo<ValueType, IndexType>::create(exec_master);
+        *host_bccoo = *this;
+        exec_master->run(bccoo::make_mem_size_bccoo(
+            host_bccoo.get(), compress_res, block_size_res, &mem_size_res));
+        auto tmp = Bccoo<ValueType, IndexType>::create(
+            exec_master, host_bccoo->get_size(), num_stored_elements,
+            block_size_res, mem_size_res, compress_res);
+        exec_master->run(
+            bccoo::make_convert_to_bccoo(host_bccoo.get(), tmp.get()));
+        *result = std::move(*tmp);
+    }
+}
+
+
+template <typename ValueType, typename IndexType>
+void Bccoo<ValueType, IndexType>::move_to(Bccoo<ValueType, IndexType>* result)
+{
+    this->convert_to(result);
+}
+
+
+template <typename ValueType, typename IndexType>
+void Bccoo<ValueType, IndexType>::convert_to(
+    Bccoo<next_precision<ValueType>, IndexType>* result) const
+{
+    using new_precision = next_precision<ValueType>;
+
+    // Definition of executors
+    auto exec = this->get_executor();
+    auto exec_master = exec->get_master();
+
+    // Getting compression and block_size values from parameters
+    bccoo::compression compress_src = this->get_compression();
+    IndexType block_size_src = this->get_block_size();
+    bccoo::compression compress_res = result->get_compression();
+    IndexType block_size_res = result->get_block_size();
+
+    // Adapting the compression and block_size values for the result
+    if ((compress_res == bccoo::compression::def_value) &&
+        (block_size_res == 0)) {
+        // For non initialized result objects, the compression and
+        // block_size values are copied from "this"
+        compress_res = compress_src;
+        block_size_res = block_size_src;
+    } else {
+        // For partial non initialized result objects, compression or
+        // block_size defaults are used
+        if (compress_res == bccoo::compression::def_value) {
+            exec->run(bccoo::make_get_default_compression(&compress_res));
+        }
+        if (block_size_res == 0) {
+            exec->run(bccoo::make_get_default_block_size(&block_size_res));
+        }
+    }
+
+    // Creating the result
+    auto num_stored_elements = this->get_num_stored_elements();
+    size_type mem_size_res{};
+    if ((compress_src == bccoo::compression::def_value) ||
+        (block_size_src == 0)) {
+        // In case of the source matrix was empty, the result is also empty
+        auto tmp = Bccoo<new_precision, IndexType>::create(exec, block_size_res,
+                                                           compress_res);
+        *result = std::move(*tmp);
+    } else if (exec == exec_master) {
+        exec->run(bccoo::make_mem_size_bccoo(this, compress_res, block_size_res,
+                                             &mem_size_res));
+        mem_size_res += num_stored_elements * sizeof(new_precision);
+        mem_size_res -= num_stored_elements * sizeof(ValueType);
```

Comment:
num_stored_elements contains the number of bytes of "this". The way to compute the size of the new object is to subtract the space occupied by the values of "this" and to add the space for the new values. To avoid negative values, the order of operations is done in reverse